android/app/src/main/java/com/vgs/android/vgs_cardform_demo/VGS_CardFormActivity.java

```diff
@@ -0,0 +1,125 @@
+package com.vgs.android.vgs_cardform_demo;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.MenuItem;
+import android.view.WindowManager;
+import android.widget.Toast;
+
+import com.braintreepayments.cardform.OnCardFormSubmitListener;
+import com.braintreepayments.cardform.utils.CardType;
+import com.braintreepayments.cardform.view.CardEditText;
+import com.braintreepayments.cardform.view.CardForm;
+import com.braintreepayments.cardform.view.SupportedCardTypesView;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class VGS_CardFormActivity extends AppCompatActivity implements OnCardFormSubmitListener,
+        CardEditText.OnCardTypeChangedListener {
+
+    private static final CardType[] SUPPORTED_CARD_TYPES = {CardType.VISA, CardType.MASTERCARD, CardType.DISCOVER,
+            CardType.AMEX, CardType.DINERS_CLUB, CardType.JCB, CardType.MAESTRO, CardType.UNIONPAY};
+
+    private SupportedCardTypesView mSupportedCardTypesView;
+
+    protected CardForm mCardForm;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.card_form);
+
+        mSupportedCardTypesView = findViewById(R.id.supported_card_types);
+        mSupportedCardTypesView.setSupportedCardTypes(SUPPORTED_CARD_TYPES);
+
+        mCardForm = findViewById(R.id.card_form);
+        mCardForm.cardRequired(true)
+                .maskCardNumber(true)
+                .maskCvv(true)
+                .expirationRequired(true)
+                .cvvRequired(true)
+                .postalCodeRequired(true)
+                .mobileNumberRequired(true)
+                .mobileNumberExplanation("Make sure SMS is enabled for this mobile number")
+                .actionLabel(getString(R.string.purchase))
+                .setup(this);
+
+        mCardForm.setOnCardFormSubmitListener(this);
+        mCardForm.setOnCardTypeChangedListener(this);
+
+        // Warning: this is for development purposes only and should never be done outside of this example app.
+        // Failure to set FLAG_SECURE exposes your app to screenshots allowing other apps to steal card information.
+        getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE);
+    }
+
+    @Override
+    public void onCardTypeChanged(CardType cardType) {
+        if (cardType == CardType.EMPTY) {
+            mSupportedCardTypesView.setSupportedCardTypes(SUPPORTED_CARD_TYPES);
+        } else {
+            mSupportedCardTypesView.setSelected(cardType);
+        }
+    }
+
+    @Override
+    public void onCardFormSubmit() {
+
+
+        URL url = null;
+        try {
+            //todo: get this property from another means
+            url = new URL("https://tntlvnzzqsz.SANDBOX.verygoodproxy.com");
```

Comment:
environment variable like in other examples https://github.com/verygoodsecurity/vgs-proxy-examples/blob/master/java/src/test/java/com/verygoodsecurity/example/ExampleTest.java#L31